docs(style-dictionary): get started (#2981)
AmadeusITGroup/otter: packages/@o3r/style-dictionary/README.md

This package exposes a set of **Hooks** and **Modules** for the [Style Dictionary](https://styledictionary.com/) to enhance the capabilities of Design Tokens.

## Get Started

Setup your [Style Dictionary](https://styledictionary.com/) in your project thanks to the command:

```shell
ng add @o3r/style-dictionary
```

> Per default the command will do the following updates:

Reviewer: ```suggestion
By default, the command will do the following updates:
```

> - Add a dev dependency to [Style Dictionary package](https://www.npmjs.com/package/style-dictionary).

Reviewer: ```suggestion
- Add a dev dependency to the [Style Dictionary package](https://www.npmjs.com/package/style-dictionary).
```

> - Create a minimal [config.mjs](https://styledictionary.com/reference/config/#_top) configuration file with the required setup for CSS and Metadata generation.

Reviewer 2: I think it's a bit late for a comment but this filename is not a bit too generic ?

Author: It is actually a question ask in the previous PR.
The name of the file is following the requirement and the default configuration name from style dictionary.
cf. https://styledictionary.com/reference/config

> - The `generate:theme` and `generate:metadata` scripts in the project *package.json*.

Reviewer 3: ```suggestion
- Add the `generate:theme` and `generate:metadata` scripts in the project *package.json*.
```

> Then you will be able to customize this setup with your project specificities:
> 
> - Customize the [Style Dictionary configuration](https://styledictionary.com/reference/config/#_top) in the generated *config.mjs*
> - Add [extensions](https://tr.designtokens.org/format/#extensions) to your source Design Token via the [enhancement mechanism](#enhancement);
> - Configure outputted files thanks to [`getTargetFiles` helper](#gettargetfiles)

Reviewer: ```suggestion
You will then be able to customize this setup with your project's specifications:

- Customize the [Style Dictionary configuration](https://styledictionary.com/reference/config/#_top) in the generated *config.mjs* file.
- Add [extensions](https://tr.designtokens.org/format/#extensions) to your source Design Token via the [enhancement mechanism](#enhancement).
- Configure output files thanks to the [`getTargetFiles` helper](#gettargetfiles).
```

}
```

> ### Available Otter extensions

Reviewer: ```suggestion
### Available Otter Extensions
```
(to be consistent with other titles that are all capitalized)

  plateforms: {
    css: {
      files: [
        ...getTargetFiles({ rule, { format: 'css', defaultFile: 'default-file.css' } })
      ]
    }
  }
});

register(sd); // Register all Otter modules
```

> [!NOTE]

> > The option `format` will be applied to all the files provided to the `getTargetFiles` function (including `defaultFile`).
> > `defaultFile` defines the default file where generate the variables not matching any rules.

Reviewer: ```suggestion
> The `format` option will be applied to all the files provided to the `getTargetFiles` function (including `defaultFile`).
> `defaultFile` defines the default file where variables not matching any rule will be generated.
```

> ## Advance

Reviewer 3: ```suggestion
## Advanced
```
?
